cpp/dolfinx/fem/CoordinateElement.h

```diff
@@ -57,6 +57,12 @@ class CoordinateElement
   /// Return the dof layout
   const ElementDofLayout& dof_layout() const;
 
+  /// Absolute increment stopping criterium for non-affine Newton solver
+  double non_affine_atol = 1.0e-8;
```

Comment:
I'd prefer to see these passed at arguments to the relevant function and keep `CoordinateElement` stateless.

Reply:
`CoordinateElement::compute_reference_geometry` is all based on reading the state of underlying CoordinateElement.
I though about the pure API approach too, but this means:
1. passing these parameters into `write_function()`, yuck, or
2. breaking the IO interface to work on lower level structures.

Probably the better approach would be 2., but it is an incomparable amount of work. See this PR as a hotfix for users to use any real-world hex meshes in IO.

Reply:
I can add the `FIXME` note if that helps?

Reply:
This feels like a top down (wrong) rather than a bottom-up fix (right). I don't mind seeing it merged now if it gets fixed later. It does seem to come back to the issue of magic interpolation behind the scenes with file output for visualisation.

Reply:
I agree with you. I can alternatively just tweak the current hardcoded tolerance and number of iterations to be more robust?

Reply:
See issue #1245.